node/src/test_utils/recorder_stop_conditions.rs

```diff
@@ -84,43 +107,43 @@ impl StopConditions {
     }
 }
 
-impl StopCondition {
-    fn resolve_condition<T: ForcedMatchable<T> + Send + 'static>(&self, msg: &T) -> bool {
+impl MsgIdentification {
+    pub fn resolve_condition<Msg: ForcedMatchable<Msg> + Send + 'static>(&self, msg: &Msg) -> bool {
         match self {
-            StopCondition::StopOnType(type_id) => Self::matches_stop_on_type::<T>(msg, *type_id),
-            StopCondition::StopOnMatch { exemplar } => {
-                Self::matches_stop_on_match::<T>(exemplar, msg)
+            MsgIdentification::ByType(type_id) => Self::matches_by_type::<Msg>(msg, *type_id),
+            MsgIdentification::ByMatch { exemplar } => {
+                Self::matches_completely::<Msg>(exemplar, msg)
```

Comment:
Maybe you can say `is_identical` instead of `matches_completely` and another optional change for `matches_by_type` be `has_the_same_type`.